packages/@sanity/schema/src/legacy/searchConfig/resolve.ts

```diff
@@ -76,14 +89,16 @@ export function deriveFromPreview(type: {
       weight: PREVIEW_FIELD_WEIGHT_MAP[fieldName],
       path: select[fieldName].split('.'),
     }))
+    .filter((spec) => spec.path.length <= maxDepth)
```

Comment:
Don't we need to subtract 1 since maxDepth here referes to the depth where 0 is the first element. Also check for `maxDepth !== -1`? 🤔 
```suggestion
    .filter((spec) => maxDepth === -1 || spec.path.length - 1 <= maxDepth)
```

Reply:
Good catch, thanks! Added tests to catch these.